sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowingStrategies.java

```diff
@@ -212,14 +230,18 @@ public static ClosingBehavior fromProto(RunnerApi.ClosingBehavior proto) {
    * the provided components to dereferences identifiers found in the proto.
    */
   public static WindowingStrategy<?, ?> fromProto(
-      RunnerApi.WindowingStrategy proto, RunnerApi.Components components) {
+      RunnerApi.WindowingStrategy proto, RunnerApi.Components components)
+      throws InvalidProtocolBufferException {
+
     Object deserializedWindowFn =
         SerializableUtils.deserializeFromByteArray(
             components
                 .getFunctionSpecsMap()
                 .get(proto.getFnId())
-                .getSdkFnSpec()
-                .getData()
+                .getSpec()
+                .getParameter()
+                .unpack(BytesValue.class)
```

Comment:
This is expecting strictly `CUSTOM_WINDOWFN_URN` - so it should be gated behind a `checkArgument` with a good message until we support anything else.